src/commands/deploy/deployTrialApp.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ProgressLocation, window, workspace, WorkspaceFolder } from 'vscode';
+import { localGitDeploy } from 'vscode-azureappservice';
+import { IActionContext } from 'vscode-azureextensionui';
+import { WebAppTreeItem } from '../../../extension.bundle';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { TrialAppTreeItem } from '../../explorer/trialApp/TrialAppTreeItem';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { selectWorkspaceFolder } from '../../utils/workspace';
+import { showDeployCompletedMessage } from './showDeployCompletedMessage';
+
+export async function deployTrialApp(context: IActionContext, trialAppTreeItem: TrialAppTreeItem): Promise<void> {
+    const workspaceFolders: WorkspaceFolder[] | undefined = workspace.workspaceFolders;
+
+    let path: string;
+    let commit: boolean = false;
+
+    if (workspaceFolders?.length === 1) {
+        path = workspaceFolders[0].uri.fsPath;
+        commit = workspaceFolders[0].name === trialAppTreeItem.metadata.siteName;
+    } else {
+        path = await selectWorkspaceFolder('Select folder containing a repository to deploy');
+        const workspaceFolder: WorkspaceFolder | undefined = workspace.getWorkspaceFolder(vscode.Uri.file(path));
```

Comment:
Should be able to move this out of the if/else clause so you only have to assign it once.